rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,120 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC: Data Groups: Processing Datasets that Span Multiple Bundles
+
+## Summary
+This RFC proposes a solution to allow processing of datasets that span multiple bundles. It addresses the general problem that the current DCP data model contains no representation of data set grouping and version-completeness beyond that of individual data bundles.  The current DCP model is restricted in that all data processing can be a one-to-one linear sequence of steps deriving one bundle from another.
+
+However, there are use cases where a given processing step can require input from multiple input bundles, resulting in a full directed acyclic graph (DAG) rather than a linear graph. The relevant DAG sub-graph must be defined and all of its associated data available to correctly initiate processing.  A notion and implemented representation of data completeness is required to support DAG processing.
+
+This RFC defines a general mechanism for grouping bundles, here called *data groups*. By providing a general method for grouping bundles, this proposal provides a mechanism for addressing various tasks that cross bundle boundaries.  This has impact on analysis ,pipelines, algorithmic complexity of data processing, data consistency, and data set quality control.  The current bundle grouping, described in detail here, is a submission to a project.  However, the concept generalizes in a manner that other groupings can be defined to accommodate future needs.
+
+## Author(s)
+- [Nick Barkas](mailto:barkasn@broadinstitute.org)
+- [Mark Diekhans](mailto:markd@ucsc.edu)
+
+## Shepherd
+- [Nick Barkas](mailto:barkasn@broadinstitute.org)
+
+## Motivation
+The current DCP model of processing has the scope of a single data bundle, which contains a single assay or analysis.  There is no mechanism to know when a group of bundles that need to be processed together is complete and available for further processing.  This results in the limitation that data processing is linear and is entirely based on ingestion packaging resulting in multiple problems, as outlined below.
```

Comment:
On "completeness"... how does this align with the concept of completeness presented in the Project Completeness RFC (https://github.com/HumanCellAtlas/dcp-community/pull/115).

Reply:
This is a low-level mechanism that is used to specify when a grouping of data is complete.  This could (hopefully would) be used to implement the tools and policy in RFC PR #115 .

If the refinement of #115 creates new requirements, this RFC will need to be amended.

Reply:
Sounds good! Thanks for the explanation!